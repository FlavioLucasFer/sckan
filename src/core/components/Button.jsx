import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { uniqueId } from 'lodash';

import Icon from './Icon';

const elemId = uniqueId('button-');

Button.propTypes = {
	title: PropTypes.string,
	color: PropTypes.string,
	size: PropTypes.string,
	type: PropTypes.string,
	className: PropTypes.string,
	icon: PropTypes.object,
	tooltipPosition: PropTypes.string,
	tooltip: PropTypes.string,
	onClick: PropTypes.func,
};

function Button(props) {
	const {
		title,
		color,
		size,
		type,
		className,
		icon,
		tooltipPosition,
		tooltip,
		onClick,
		...rest
	} = props;

	
	useEffect(() => {
		M.Tooltip.init(document.querySelectorAll('.tooltipped'));
	}, []);

	function handleClick() {
		const elems = document.querySelectorAll('.tooltipped');

		for (let i = 0; i < elems.length; i++) {
			const tooltipInstance = M.Tooltip.getInstance(elems[i]);
			tooltipInstance.close();
		}

		if (onClick) {
			onClick();
		}
	}

	function renderIcon() {
		if (!icon)
			return null;

		return (
			<Icon {...icon}>
				{icon.name}
			</Icon>
		);
	}

	return (
		<a {...rest} 
			id={elemId}
			className={`
				${type ? `btn-${type}` : 'btn'}
				${size ? `btn-${size}` : 'btn-medium'} 
				${tooltip && 'tooltipped'}
				waves-effect waves-light 
				${color || 'white'}
				${className || ''}
			`}
			data-position={tooltipPosition || 'bottom'}
			data-tooltip={tooltip || ''}
			onClick={handleClick}>
			{renderIcon()}
			{title}
		</a>
	);
}

export default Button;
