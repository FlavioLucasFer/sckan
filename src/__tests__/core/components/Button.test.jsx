import React from 'react';
import { shallow } from 'enzyme';

import Button from 'core/components/Button';

describe('Testing Button core component', () => {
	it('should render correctly', () => {
		const wrapper = shallow(
			<Button />
		);

		expect(wrapper.debug()).toMatchSnapshot();
		
		wrapper.setProps({
			title: 'testing',
			className: 'grey-text',
			type: 'round',
			color: 'red',
			size: 'small',
			tooltipPosition: 'top',
			tooltip: 'testing',
			icon: {
			name: 'business',
			},
			onClick: () => console.log('testing button'),
		});

		expect(wrapper.debug()).toMatchSnapshot();
	});
});