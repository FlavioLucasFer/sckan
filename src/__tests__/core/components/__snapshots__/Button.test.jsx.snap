// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button core component should render correctly 1`] = `"<a id=\\"button-1\\" className=\\"\\\\n\\\\t\\\\t\\\\t\\\\tbtn\\\\n\\\\t\\\\t\\\\t\\\\tbtn-medium \\\\n\\\\t\\\\t\\\\t\\\\tundefined\\\\n\\\\t\\\\t\\\\t\\\\twaves-effect waves-light \\\\n\\\\t\\\\t\\\\t\\\\twhite\\\\n\\\\t\\\\t\\\\t\\\\t\\\\n\\\\t\\\\t\\\\t\\" data-position=\\"bottom\\" data-tooltip=\\"\\" onClick={[Function: handleClick]} />"`;

exports[`Testing Button core component should render correctly 2`] = `
"<a id=\\"button-1\\" className=\\"\\\\n\\\\t\\\\t\\\\t\\\\tbtn-round\\\\n\\\\t\\\\t\\\\t\\\\tbtn-small \\\\n\\\\t\\\\t\\\\t\\\\ttooltipped\\\\n\\\\t\\\\t\\\\t\\\\twaves-effect waves-light \\\\n\\\\t\\\\t\\\\t\\\\tred\\\\n\\\\t\\\\t\\\\t\\\\tgrey-text\\\\n\\\\t\\\\t\\\\t\\" data-position=\\"top\\" data-tooltip=\\"testing\\" onClick={[Function: handleClick]}>
  <Icon name=\\"business\\">
    business
  </Icon>
  testing
</a>"
`;
